### PR TITLE
core: introduce %R specifier for pretty hostname

### PR DESCRIPTION
**Pull request by mrc0mmand**
Resolves: #20054

### CODE REVIEW

File: src/shared/specifier.c [hunk 1]
```diff
@@ -208,6 +208,21 @@ int specifier_short_host_name(char specifier, const void *data, const char *root
         return 0;
 }
 
+int specifier_pretty_host_name(char specifier, const void *data, const char *root, const void *userdata, char **ret) {
+        char *n = NULL;
+        int r;
+
```
Comment (bluca) [inline src/shared/specifier.c:228]: `assert(ret)`

Comment (mrc0mmand) [reply src/shared/specifier.c:228]: Given that almost none of the specifier helper functions assert on `ret`, it's been either left out intentionally or should be added everywhere :-)

**Review by bluca — APPROVED** [review]

File: src/shared/specifier.c [hunk 2]
```diff
@@ -208,6 +208,23 @@ int specifier_short_host_name(char specifier, const void *data, const char *root
...
+        assert(ret);
+
+        r = get_pretty_hostname(&n);
+        if (r < 0) {
+                n = strdup("(none)");
```
Comment (yuwata) [inline src/shared/specifier.c]: Not sure, but if the pretty hostname is not set, then it is better to fallback the short hostname??

Also, you can drop the assignment to `r`.

Comment (mrc0mmand) [reply src/shared/specifier.c]: That definitely sounds more sensible, updated, thanks!

File: man/standard-specifiers.xml [hunk 3]
```diff
@@ -34,6 +34,11 @@
     <entry>Short host name</entry>
     <entry>The hostname of the running system, truncated at the first dot to remove any domain component.</entry>
   </row>
+  <row id='R'>
+    <entry><literal>%R</literal></entry>
+    <entry>Pretty host name</entry>
+    <entry>The pretty hostname of the running system. If not set, resolves to the short hostname.</entry>
```
Comment (poettering) [inline man/standard-specifiers.xml]: maybe add a comment that this is PRETTY_HOSTNAME from machine-info(5).

Comment (mrc0mmand) [reply man/standard-specifiers.xml]: Makes sense, updated.

**Pull request merged**